core/text/src/markdown/lexer.rs

```diff
@@ -0,0 +1,215 @@
+use nom::{IResult, InputLength, InputTake, InputIter, Slice};
+use std::ops::{Range, RangeFrom, RangeTo, RangeFull};
+use std::slice::Iter;
+use std::iter::Enumerate;
+use nom::multi::*;
+use nom::branch::*;
+use nom::combinator::*;
+use nom::bytes::complete::*;
+use nom::character::complete::*;
+use nom::sequence::*;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum LexToken {
+    ControlWordStarter,
+    LBrace,
+    RBrace,
+    Text(String),
```

Comment:
This is not part of the lexing
Tokens are the following
- WhiteSpace
- Word
- LBrace
- RBrace
- ControlWordStarter

Text is an internal literal